push-athena-data-to-rds/push-athena-data-to-rds.go

```diff
@@ -14,76 +15,76 @@ import (
 	"github.com/aws/aws-lambda-go/lambda"
 )
 
-func getAssettypeDataFromAthena(athenaDBName string, athenaTableName string, s3Location string, queryParams map[string]string) {
-	completeS3Location := utils.CompleteS3Location(s3Location, "assettype", queryParams)
-
-	query, queryErrMsg := athena.AssetypeDataQuery(queryParams, athenaDBName, athenaTableName)
-
-	if queryErrMsg.Err != nil {
-		fmt.Println(queryErrMsg.Message, queryErrMsg.Err)
-	}
-
-	s3FileName, athenaErrMsg := athena.SaveDataToS3(query, athenaDBName, completeS3Location)
-
-	if athenaErrMsg.Err != nil {
-		fmt.Println(athenaErrMsg.Message, athenaErrMsg.Err)
+func getQuery(dataSource string, queryParams map[string]string, athenaDBName string, athenaTableName string) (string, types.ErrorMessage) {
+	switch dataSource {
+	case "assettype":
+		return athena.AssetypeDataQuery(queryParams, athenaDBName, athenaTableName)
+	case "host":
+		return athena.PrimaryDomainDataQuery(queryParams, athenaDBName, athenaTableName)
+	case "varnish":
+		return athena.VarnishDataQuery(queryParams, athenaDBName, athenaTableName)
+	case "frontend_haproxy":
+		return athena.FrontendHaproxyDataQuery(queryParams, athenaDBName, athenaTableName)
+	case "gumlet":
+		return athena.GumletDataQuery(queryParams, athenaDBName, athenaTableName)
+	default:
+		return "", types.ErrorMessage{
+			Message: "Wrong DataSource",
+			Err:     nil,
+		}
 	}
-
-	api.SaveAthenaData(s3FileName, queryParams, "assettype")
 }
 
-func getStatsOnPrimaryDomainFromAthena(athenaDBName string, athenaTableName string, s3Location string, queryParams map[string]string) {
-	completeS3Location := utils.CompleteS3Location(s3Location, "host", queryParams)
-
-	query, queryErrMsg := athena.PrimaryDomainDataQuery(queryParams, athenaDBName, athenaTableName)
-
-	if queryErrMsg.Err != nil {
-		fmt.Println(queryErrMsg.Message, queryErrMsg.Err)
+func getDBName(dataSource string) string {
+	switch dataSource {
+	case "assettype":
+		return os.Getenv("CLOUDFLARE_DB")
+	case "host":
+		return os.Getenv("CLOUDFLARE_DB")
+	case "varnish":
+		return os.Getenv("ALB_DB")
+	case "frontend_haproxy":
+		return os.Getenv("ALB_DB")
+	case "gumlet":
+		return os.Getenv("GUMLET_DB")
+	default:
+		return "Wrong DataSource"
 	}
-
-	s3FileName, athenaErrMsg := athena.SaveDataToS3(query, athenaDBName, completeS3Location)
-
-	if athenaErrMsg.Err != nil {
-		fmt.Println(athenaErrMsg.Message, athenaErrMsg.Err)
-	}
-
-	api.SaveAthenaData(s3FileName, queryParams, "host")
 }
 
-func getVarnishDataFromAthena(athenaDBName string, athenaTableName string, s3Location string, queryParams map[string]string) {
-	completeS3Location := utils.CompleteS3Location(s3Location, "varnish", queryParams)
-
-	query, queryErrMsg := athena.VarnishDataQuery(queryParams, athenaDBName, athenaTableName)
-
-	if queryErrMsg.Err != nil {
-		fmt.Println(queryErrMsg.Message, queryErrMsg.Err)
+func getTableName(dataSource string) string {
```

Comment:
I see the same switchCase here nearly three times. At some point, you sould replace this with

function getConfig(datasource string) {
  return { table-name: xyz, database: xyz, query: Query}
}